### DISCUSSION TITLE
Garage door with dry contact switch

### DISCUSSION THREAD

**Discussion by childledj** [RavenSystem/esp-homekit-devices#1586] [MEPLHAA Scripts]
Hello to everyone!
I have Sonoff Basic remaked to Dry contact
and I want to add it parallel with physical button.
I have no wired sensors attached to Sonoff but have separate Zigbee sensor that indicates the garage door is closed.
I absolutely dumb with json config so can anybody help me with config??
I need stateless switch that indicates as garage door

Is it possible to bing garage door switch and separate sensor in Apple Homekit??

**Comment by Daveblanche** [comment]
Yes!
I have the same set up. 
This is the JSON. Set up to inch for 5 seconds, from memory. I left it as a switch as I use the sensor itself as the garage door indicator, otherwise I ended up with two garage doors on the garage page on HomeKit.  
This might work for you. 
It’s from the database. 

{"c":{"l": 13,"b":[{"g":0,"t":5}]},"a":[{"t":1,"i":5,"d": 10,"s":5,"0":{"r":[{"g":12,"v":0}]},"1":{"r":[{"g":12,"v":1}]},"b":[{"g":0}]}]}

**Reply by childledj** [reply]
Thank you! But have you tried to bind sensor with switch (if it possible) when you have two garage doors in Home app?
Now I have used this config from database - it indicates as lock, but it sends two Notification - first when button pushed (i.e. Garage door unlocked) and second after 2 seconds (i.e. Garage door locked)
{"c":{"0":{"r":[{"g":12}]},"l":13,"b":[{"g":0,"t":5}]},"a":[{"0":{"r":[{"g":12,"i":1},{"g":12,"v":1,"i":2}]},"t":4,"xa":0,"b":[{"g":0,"t":1}],"s":1,"i":4,"f0":[]}]}

**Comment by Daveblanche** [comment]
No need to bind. The sensor says if the door is open or closed. The door (in my case) has its own method of sensing when it is closed or opened, and the switch just inches for five seconds whenever it is pressed such that if it is closed it will open or if it’s next action is to close then it will close.

**Comment by cristidobrescu** [comment]
Hi @childledj @Daveblanche 
Could someone of you help me. I also have an sonoff basic remarked for dry contact and I would like to know where to soldering two wires for contact sensor on the main unit, i can’t figured it out. I have to mention my unit is sonoff basic r2 without GPIO14
![Uploading 90603065-AC30-4DDA-9E40-60E8E2701CE3.jpeg…]()

**Comment by cristidobrescu** [comment]
![F8EDEFBB-F159-477E-9CF1-EB9557FC8113](https://user-images.githubusercontent.com/99839121/156911995-cfa741f6-ae37-47d9-8294-8861c4954b83.jpeg)
